mesonbuild/cargo/interpreter.py

```diff
@@ -156,6 +164,21 @@ class Package:
     def __post_init__(self) -> None:
         self.api = _version_to_api(self.version)
 
+    @classmethod
+    def from_raw(cls, raw: manifest.Package) -> Self:
+        pkg = T.cast('manifest.FixedPackage',
+                     {fixup_meson_varname(k): v for k, v in raw.items()})
+
+        unexpected = set(pkg) - {x.name for x in dataclasses.fields(cls)}
+        if unexpected:
+            mlog.warning("Package entry", pkg['name'], "has unexpected keys",
+                         '"{}".'.format(", ".join(sorted(unexpected))), _EXTRA_KEYS_WARNING)
+            for k in unexpected:
+                # Mypy and Pyright can't prove that this is okay
+                del pkg[k]  # type: ignore[misc]
```

Comment:
I would factor that code into a common function to avoid duplication in each from_raw() methods.